base_layer/core/src/blocks/block_header.rs

```diff
@@ -168,16 +169,34 @@ impl BlockHeader {
         }
     }
 
+    /// Returns the [std::cmp::Ordering] of the timestamp of this header compared to the "other" header
+    pub fn compare_timestamp(&self, other: &BlockHeader) -> Ordering {
+        self.timestamp.as_u64().cmp(&other.timestamp.as_u64())
+    }
+
     #[cfg(feature = "base_node")]
     pub fn into_builder(self) -> BlockBuilder {
         BlockBuilder::new(self.version).with_header(self)
     }
 
-    /// Given a slice of headers (in reverse order), calculate the maximum, minimum and average periods between them
+    /// Given a slice of headers, calculate the maximum, minimum and average periods between them.
+    /// Allocates in case the headers are in the wrong order.
```

Comment:
This function always allocates for len >= 2 (`to_vec` creates a [copy of the slice and each element](https://github.com/rust-lang/rust/blob/051d1176b786aadd7d7c048f822cb6bfab00fe03/library/alloc/src/slice.rs#L209)) - but that is fine for this function.

Perhaps add a comment that headers should be ordered from oldest to youngest.

Reply:
Yeah the wording isn't great, will update it.